src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java

```diff
@@ -140,13 +140,9 @@ public void clusterDescriptionChanged(
           if (!latch.await(latchTimeout, TimeUnit.MILLISECONDS)) {
             configValue.addErrorMessage("Unable to connect to the server.");
             mongoClient.close();
```

Comment:
Do we want such condition for Non srv protocol host (self-managed instance) as well?

Reply:
Could you try out a scenario with an invalid self-managed instance?